src/radical/pilot/agent/bootstrap_0.sh

```diff
@@ -146,6 +135,39 @@ export PYTHONNOUSERSITE=True
 # nodes), we try to convince PIP to run parallel `make`
 export MAKEFLAGS="-j"
 
+
+# ------------------------------------------------------------------------------
+#
+# check what env variables changed from the original env, and create a
+# `deactivate` script which resets the original values.
+#
+create_deactivate()
+{
+    # capture the current environment
+    env | sort | grep '=' | sed -e 's/\([^=]*\)=\(.*\)/export \1="\2"/g' \
+        > env.bs_0
+
+    # find all variables which changed
+    changed=$(diff -w env.orig env.bs_0 | grep '='        \
+                                        | cut -f 3 -d ' ' \
+                                        | cut -f 1 -d '=' \
+                                        | uniq)
+
+    # capture the original values in `deactivate`.  If no original value exists,
+    # unset the variable
+    for var in $changed
+    do
+        val=$(grep -e "^export $var=" env.orig | cut -f 2- -d '=')
+        if test -z "$val"
```

Comment:
`if ! test -z "$val"` (currently it does opposite: unsets if original values exists, and export if not)

Reply:
Ah, thanks, nice catch

Reply:
I realized that this wasn't correct anyway: an empty value does not mean that the var did not exist.  This is thus fixed differently now - can you have a look?